circle.yml

```diff
@@ -1,4 +1,17 @@
 machine:
   node:
     version: 4.3.0
-
+test:
+  override:
+    - ./node_modules/.bin/eslint --quiet:
+        parallel: true
+        files:
+          - client/**/*.js
+          - client/**/*.jsx
+          - server/**/*.js
+          - server/**/*.jsx
+    - NODE_ENV=development make build && NODE_ENV=development ./bin/run-tests:
```

Comment:
This actually gets really confusing. Some Makefile ignore an already set NODE_ENV and reset it to TEST. And run-tests sets it, but doesn't export it or pass it along, so subprocs don't seem to pick it up. 

In the current system I think we're actually using the outer node_env just to generate the config file? Would be nice to not have the switcheroo and just use an env of test everywhere.